craft_parts/executor/part_handler.py

```diff
@@ -180,10 +183,13 @@ def _run_build(self, step_info: StepInfo, *, update=False) -> StepState:
         }
         assets.update(_get_machine_manifest())
 
+        overlay_hash = self._compute_layer_hash(for_all_parts=True)
```

Comment:
can you add a code comment on why for_all_parts and what the objective of this piece of code is? Not so much the what, but why.

Reply:
Added a comment explaining the reason.